[DO NOT MERGE] CCD in prevention mode

smathangi opened the pull request:
added excluded rules based on analysis of ccd prod in detection mode (1 week detection data)
https://tools.hmcts.net/jira/browse/RDM-3807

smathangi committed:
added excluded rules based on analysis of ccd prod in detection mode (1 week data)

cakeben reviewed (approved):
I approve the changes but this should not be merged to master it should go on a different branch of the module which CCD should then seek to obtain sign off from SecOps before rolling out the change

mario-paniccia reviewed (approved):
👍

mario-paniccia commented:
> I approve the changes but this should not be merged to master it should go on a different branch of the module which CCD should then seek to obtain sign off from SecOps before rolling out the change

I seems clear to me that CCD needs to run with custom exclusion rules. I think it'll make sense for the `cnp-module-waf` module on master branch support allowing setting custom exclusion rules. Until we have that, we'll always need to be on the `ccd-waf` branch.

timja commented:
> > I approve the changes but this should not be merged to master it should go on a different branch of the module which CCD should then seek to obtain sign off from SecOps before rolling out the change
> 
> I seems clear to me that CCD needs to run with custom exclusion rules. I think it'll make sense for the `cnp-module-waf` module on master branch support allowing setting custom exclusion rules. Until we have that, we'll always need to be on the `ccd-waf` branch, which is not that good.

correct we haven't invested in this though as we plan to move away from application gateway and use front door instead which supports application waf profiles in a much better way

smathangi committed:
removed invalid ruleIds

smathangi committed:
added missing ruleId  200004

nathan-clark committed:
RDM-6863-disableRules

nathan-clark committed:
Merge pull request #49 from hmcts/RDM-6863-disableRules

RDM-6863-disableRules-RDM-6863

mario-paniccia committed:
remove offending duplicate rule and merged into existing

mario-paniccia committed:
Merge pull request #51 from hmcts/remove_offending_duplicate_rule

Remove offending duplicate rule

njrich28 committed:
Add conditional request timeout property

njrich28 committed:
Fix test for existence of field

Previously using `equals` to test for existence of field, switched to using `contains` which is correct test.

https://docs.microsoft.com/en-gb/azure/azure-resource-manager/templates/template-functions-array#contains

njrich28 committed:
Wrap the other property in conditional

MSancaktutar committed:
Merge pull request #52 from hmcts/njrich28-patch-1

Add conditional request timeout property

nathan-clark committed:
RDM-7904-disable-942300-933160

nathan-clark committed:
Merge pull request #54 from hmcts/RDM-7904-disable-942300-933160

RDM-7904-disable-942300-933160
